app/bundles/EmailBundle/Controller/EmailController.php

```diff
@@ -1637,4 +1670,102 @@ protected function getDefaultOrderDirection(): string
     {
         return 'DESC';
     }
+
+    protected function getEmailModel(): EmailModel
+    {
+        $model = $this->getModel('email');
+        \assert($model instanceof EmailModel);
+
+        return $model;
+    }
+
+    /**
+     * @return array<int|string, array<int|string, int|string>>
+     *
+     * @throws Exception
+     */
+    private function getCountriesTableData(Email $entity): array
+    {
+        $model = $this->getEmailModel();
+
+        // get A/B test information
+        $parent = $entity->getVariantParent();
+
+        // get translation parent
+        $translationParent = $entity->getTranslationParent();
+        $includeVariants   = (($entity->isVariant() && empty($parent)) || ($entity->isTranslation() && empty($translationParent)));
+
+        return $model->getCountryStats(
+            $entity,
+            $includeVariants,
+        );
+    }
+
+    /**
+     * @return array<int, string>
+     */
+    private function getCountriesTableExportHeader(): array
+    {
+        return [
+            $this->translator->trans('mautic.lead.lead.thead.country'),
+            $this->translator->trans('mautic.email.graph.line.stats.sent'),
+            $this->translator->trans('mautic.email.graph.line.stats.read'),
+            $this->translator->trans('mautic.email.clicked'),
+        ];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function countryStatsAction(
+        int $objectId
```

Comment:
```suggestion
        EmailModel $model,
        int $objectId
```

You can use dependency injection in the controller method. After this change you don't need to modify the constructor or define the `getEmailModel` method that checks the type.